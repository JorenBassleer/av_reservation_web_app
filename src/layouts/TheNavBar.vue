<template>
  <nav>
    <router-link to="/">
      HOME
    </router-link> |
    <router-link to="/apparaten">
      APPARATEN
    </router-link> |
    <router-link to="/mijn-reservatie">
      MIJN RESERVATIE <span id="appliances-number">{{ amountOfAppliances }}</span>
    </router-link>
  </nav>
</template>
<script setup>
import { useStore } from 'vuex';
import { computed } from 'vue';

const store = useStore();

const amountOfAppliances = computed(() => store.getters.getTotalAmountOfAppliancesInReservation);
</script>
<style scoped>
nav {
  padding: 30px;
}

nav a {
    font-size: 20px;
    text-decoration: none;
    font-weight: bold;
    color: #2c3e50;
}
nav a:hover {
    text-decoration: underline;
}
nav a.router-link-exact-active {
  color: white;
}
</style>
