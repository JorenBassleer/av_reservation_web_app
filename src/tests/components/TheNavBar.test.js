/**
 * @vitest-environment
 */
import { mount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import {
  expect, describe, it, beforeEach,
} from 'vitest';
import TheNavBar from '../../layouts/TheNavBar.vue';

describe('Appliance.vue', () => {
  let store;
  let wrapper;
  beforeEach(() => {
    store = createStore({
      getters: {
        getTotalAmountOfAppliancesInReservation() { return 60; },
      },
    });
    wrapper = mount(TheNavBar, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });
  });
  it('Test router-links', () => {
    expect(wrapper.findAllComponents(RouterLinkStub)[0].props().to).toBe('/');
    expect(wrapper.findAllComponents(RouterLinkStub)[1].props().to).toBe('/apparaten');
    expect(wrapper.findAllComponents(RouterLinkStub)[2].props().to).toBe('/mijn-reservatie');
  });
  it('Match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('#appliances-number has total amount of appliances in reservation', async () => {
    // Arrange
    // Act
    const applianceNumber = parseInt(wrapper.find('#appliances-number').text(), 10);
    // 60 like we defined in our mock store
    // Assert
    expect(applianceNumber).toBe(60);
  });
});
