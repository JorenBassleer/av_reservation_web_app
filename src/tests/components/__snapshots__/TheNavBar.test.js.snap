// Vitest Snapshot v1

exports[`Appliance.vue > Match snapshot 1`] = `"<nav data-v-517c5da5=\\"\\"><a data-v-517c5da5=\\"\\"> HOME </a> | <a data-v-517c5da5=\\"\\"> APPARATEN </a> | <a data-v-517c5da5=\\"\\"> MIJN RESERVATIE <span id=\\"appliances-number\\" data-v-517c5da5=\\"\\">60</span></a></nav>"`;
